### DISCUSSION TITLE
Support for rgba() or hsla() in waveColor?

### DISCUSSION THREAD

**Discussion by equilet** [katspaugh/wavesurfer.js#3828] [General]
Hello,

I'm wondering if it would be possible to implement something like `rgba(100,100,100,0.)` or `hsla(0,100,50,0.)` for the `waveColor` property of the creation method.  My reason is that I'd like to have multiple instances of the same ws on different virtual tracks, some transparent and some not.  I can fake this by setting the waveform color to the same color as the background, but it's not exactly what I'm looking for.  I realize my case is exotic– I'm building a system that has annotations (regions) and running out of space in the single "lane" that WS offers.

Thanks for any thoughts!
I'm sure there may be something obvious that I'm missing so feel free to recommend an alternative approach.  :)

**Comment by katspaugh** [comment]
You can pass anything that a [Canvas fillStyle](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/fillStyle) accepts, including [rgba and hsl colors](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value).

**Comment by equilet** [comment]
Strange - when I issue `rgba(0,0,0,0.)` I get black.  I tried `rgba(0,0,0,0.001)` and it gives me what I want.  Thank you for verifying- I think it may have more to do with canvas browser integration than wavesurfer itself...

**Reply by katspaugh** [reply]
Works for me with rgba(0, 0, 0, 0) (no dot!). Or you can just type “transparent”.

**Reply by equilet** [reply]
thank you!

**Comment by equilet** [comment]
btw if you were implementing something similar, would you recommend that I include `regions` instances that refer to a single wavesurfer instance, or should I be spawning separate (duplicated) WS instances that regions attach to individually?

**Reply by katspaugh** [reply]
I don’t understand your use case 100% but generally one wavesurfer instance = one audio track and one waveform on which regions are overlayed. You can’t reuse the same instance for different tracks.

**Reply by equilet** [reply]
I think your interpretation is sufficient without getting into the details.  Many thanks for your quick reply!
